File: opracowaniaAlgo/RBT/RBTcodes/java/nodeClass.java
public class Node<T extends Comparable<T>>
{}
    public Node<T> parent,left,right;
    public T data;
    public boolean isBlack;
    
    public Node() {}
    public Node(T data)
    {
        this.data=data;
        this.isBlack=false;
    }


}

File: opracowaniaAlgo/RBT/RBTcodes/cpp/nodeClass.cpp
template <class T>
class Node
{
	template <class T> friend class RedBlackTree;

	Node<T>* parent, * left, * right;
	T data;
	bool isBlack;

public:
	Node() {}
	Node(T data)
	{
		this->data = data;
		this->isBlack = false;
	}
};